code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -963,8 +963,33 @@
 	color = "#A0A0A0" //mannitol is light grey, neurine is lighter grey"
 	chem_flags = CHEMICAL_RNG_GENERAL | CHEMICAL_RNG_FUN | CHEMICAL_RNG_BOTANY | CHEMICAL_GOAL_CHEMIST_BLOODSTREAM | CHEMICAL_GOAL_BOTANIST_HARVEST
 
+/datum/reagent/medicine/mannitol/on_mob_add(mob/living/carbon/C)
+	if(HAS_TRAIT(C, TRAIT_BRAIN_TUMOR))
+		overdose_threshold = 35 // special overdose to brain tumor quirker
+	..()
+
+
 /datum/reagent/medicine/mannitol/on_mob_life(mob/living/carbon/C)
-	C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2*REM)
+	if(HAS_TRAIT(C, TRAIT_BRAIN_TUMOR)) // to brain tumor quirker
+		SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "brain_tumor", /datum/mood_event/brain_tumor_mannitol)
+		if(!overdosed)
+			C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -0.5*REM)
+	else // to ordinary people
+		C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2*REM)
+	..()
+
+/datum/reagent/medicine/mannitol/overdose_process(mob/living/carbon/C)
+	if(HAS_TRAIT(C, TRAIT_BRAIN_TUMOR))
+		if(prob(10))
+			C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -0.1*REM)
+	..()
+
+/datum/reagent/medicine/mannitol/on_mob_end_metabolize(mob/living/carbon/C)
+	if(HAS_TRAIT(C, TRAIT_BRAIN_TUMOR))
+		// re-send the mood signal. to make the time measure unreliable but within an estimation.
+		SEND_SIGNAL(C, COMSIG_CLEAR_MOOD_EVENT, "brain_tumor")
+		SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "brain_tumor", /datum/mood_event/brain_tumor_mannitol)
```

Comment:
Not sure if this is wrong, or line 974 is but there is an inconsistency which indicates that one of them is done wrong.
On line 974 you send the signal to add the mood event, but don't clear it before hand.
When you do it here you clear the mood event and then send the signal to add the mood event again.

If you need to clear the mood event in order to reset the timer, then line 974 will need to be updated because otherwise you can have the event timeout while you still have mannitol in your system. If not, then the clear mood event can be removed from this line.

Reply:
I thought the mood timer isn't properly calculated upon reapply at my previous test. I tested it again, and I found out I was wrong - `on_mob_end_metabolize()` isn't needed, and sending the same mood signal always refreshes the mood event's timer to `timeout` value which is randomly declared from `new()`

Although, I am not still sure where I should put `SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "brain_tumor", /datum/mood_event/brain_tumor_mannitol)` at best. medicine dm? negative dm?